### PR TITLE
Add search

### LINKED ISSUE #69: Search function
On the desktop client, I mostly use the search function to find credentials. I think a search function would improve the mobile app as well.

### PR DESCRIPTION
**Pull request by chkichkiparmparm**
Hey guys,
This is a PR for issue #69 Search function for mobile. I've added the search option to the sheets in both the archive list page, and the groups pages. Both take you to the same view which allows you to search all unlocked archives.

Let me know if you have any feedback or questions.

Closes #69

### CODE REVIEW

**Comment by perry-mitchell** [comment]
This looks amazing! I'll jump in and test it shortly..

File: source/shared/entries.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            ...current,
+            [next.workspace.archive.id]: next.id
+        }),
+        {}
+    );
+    const archives = unlockedSources.map(source => source.workspace.archive);
+    const finder = new EntryFinder(archives);
+
+    return Promise.all(
+        finder.search("/" + term + "/").map(async result => {
```
Comment (perry-mitchell) [inline source/shared/entries.js]: What's with `"/" + term + "/"`?

Comment (perry-mitchell) [reply source/shared/entries.js]: Not sure that this function needs to be `async` though, as there's no `await` calls.

Comment (chkichkiparmparm) [reply source/shared/entries.js]: I misunderstood the finder.search function. I thought it was supposed to take a regex expression. Anyway, I removed those slashes and it worked fine. I also removed the async and everything worked as expected!

Comment (perry-mitchell) [reply source/shared/entries.js]: Great! Thanks @chkichkiparmparm!

**Comment by perry-mitchell** [comment]
I'll merge this into a feature to do some final checks. Thanks @chkichkiparmparm!

**Pull request merged**